Add to CI check for specification version.

kairoaraujo opened the pull request:
This commit adds to the CI an automatic check for the TUF
specification version and compares it with the python-tuf metadata
API version.

If the version does not match and there is not a issue already open,
a new issue is opened.

Closes #1598

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

- [X] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

Commit:
Add to CI check for specification version.

This commit adds to the CI an automatic check for the TUF
specification version and compares it with the python-tuf metadata
API version.

If the version does not match and there is not a issue already open,
a new issue is opened.

Closes #1598

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

kairoaraujo commented:
First, thank you to @jku  for making it so easy to do.

Some additional information:

I run the tests on my forked repo.
The [first run](https://github.com/kairoaraujo/python-tuf/runs/4740564000?check_suite_focus=true) created the issue. 
The [second run]( https://github.com/kairoaraujo/python-tuf/actions/runs/1668195526) does not open as the issue is already opened ( + there is an issue already closed.)

I decided not to add any version information to not make the issue obsolete in case of a new version.

jku commented on .github/workflows/specification-version.yml:
```diff
@@ -0,0 +1,55 @@
+            console.log("Specification version does not match...")
+            const repo = context.repo.owner + "/" + context.repo.repo
+            const issues = await github.rest.search.issuesAndPullRequests({
+              q: "specification+has+new+version+in:title+state:open+type:issue:" + repo,
```
I never spent the time to test if you can give the repo and owner as separate parameters like in `issues.create()` which would probably be best... However, embedding the repo and owner in the query definitely works but the way it's done here looks like syntax error to me: the query string ends with `"type:issue:theupdateframework/python-tuf"`. Can you check this?

kairoaraujo replied:
I didn't find specific separated parameters in the documentation.
https://docs.github.com/en/rest/reference/search#search-issues-and-pull-requests

But yes, for some reason, I removed the repository parameter. I fixed it now

Commit:
Fix query syntax

Fix query syntax that was missing the repository parameter

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

MVrachev commented:
The CI failure is fixed in https://github.com/theupdateframework/python-tuf/pull/1762.
If you rebase after the merge of this pr it should be fixed.

jku commented on .github/workflows/specification-version.yml:
```diff
@@ -0,0 +1,55 @@
+            console.log("Specification version does not match...")
+            const repo = context.repo.owner + "/" + context.repo.repo
+            const issues = await github.rest.search.issuesAndPullRequests({
+              q: "specification+has+new+version+in:title+state:open+type:issue+repo:" + repo,
```
I'm guessing the "in:title" search might actually require quotes to get an exact search string but I think I'm fine with this: let's have another look if we get false positives in this.

jku reviewed (approved):
I think this is good enough, thanks.

I'm a bit on the fence about output (both logging and the issue message): It says what was done but includes nothing that would help debugging if things break (like the actual version strings that were compared)...

So, approving as is but please have a look at the logging/output: if you think it needs work we can do that

Commit:
Improve the logs output

Minor changes to the console logs add versioning and simplify when
they are logged.

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

Commit:
Add workflows permissions

read the contents and write (open) issues

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>
